src/fetchcode/package_managers.py

```diff
@@ -0,0 +1,523 @@
+
+    for release in response["versions"]:
+        version = release["version"]
+        version = remove_debian_default_epoch(version)
+        yield PackageVersion(value=version)
+
+
+@router.route("pkg:maven/.*")
+def get_maven_versions_from_purl(purl):
+    """Fetch versions of Maven packages from Maven Central maven-metadata.xml data."""
+    purl = PackageURL.from_string(purl)
+    group_id = purl.namespace
+    artifact_id = purl.name
+    if not group_id:
+        return
+
+    group_url = group_id.replace(".", "/")
+    endpoint = (
+        f"https://repo1.maven.org/maven2/{group_url}/{artifact_id}/maven-metadata.xml"
+    )
+    response = get_response(url=endpoint, content_type="binary")
+    if response:
+        xml_resp = ET.ElementTree(ET.fromstring(response.decode("utf-8")))
+        yield from maven_extract_versions(xml_resp)
+
+
+@router.route("pkg:nuget/.*")
+def get_nuget_versions_from_purl(purl):
+    """Fetch versions of NuGet packages from the nuget.org API."""
+    purl = PackageURL.from_string(purl)
+    pkg = purl.name.lower()
+    url = f"https://api.nuget.org/v3/registration5-semver1/{pkg}/index.json"
+    resp = get_response(url=url)
+    if resp:
+        yield from nuget_extract_versions(resp)
+
+
+@router.route("pkg:composer/.*")
+def get_composer_versions_from_purl(purl):
+    """Fetch versions of PHP Composer packages from the packagist.org API."""
+    purl = PackageURL.from_string(purl)
+    if not purl.namespace:
+        return
+
+    pkg = f"{purl.namespace}/{purl.name}"
+    response = get_response(url=f"https://repo.packagist.org/p/{pkg}.json")
+    if response:
+        yield from composer_extract_versions(response, pkg)
+
+
+@router.route("pkg:hex/.*")
+def get_hex_versions_from_purl(purl):
+    """Fetch versions of Erlang packages from the hex API."""
+    purl = PackageURL.from_string(purl)
+    response = get_response(
+        url=f"https://hex.pm/api/packages/{purl.name}",
+        content_type="json",
+    )
+    for release in response["releases"]:
+        yield PackageVersion(
+            value=release["version"],
+            release_date=dateparser.parse(release["inserted_at"]),
+        )
+
+
+@router.route("pkg:conan/.*")
+def get_conan_versions_from_purl(purl):
+    """Fetch versions of ``conan`` packages from the Conan API."""
+    purl = PackageURL.from_string(purl)
+    response = get_response(
+        url=(
+            "https://raw.githubusercontent.com/conan-io/conan-center-index/"
+            f"master/recipes/{purl.name}/config.yml"
+        ),
+        content_type="yaml",
+    )
+    for version in response["versions"].keys():
+        yield PackageVersion(value=version)
+
+
+@router.route("pkg:github/.*")
+def get_github_versions_from_purl(purl):
+    """Fetch versions of ``github`` packages using GitHub REST API."""
+    purl = PackageURL.from_string(purl)
+    response = get_response(
+        url=(f"https://api.github.com/repos/{purl.namespace}/{purl.name}/releases"),
```

Comment:
Use GitHub tags when release is not available. 
For example, https://github.com/nginx/nginx/releases doesn't have any release, but it does have tags https://github.com/nginx/nginx/tags.